app/src/main/java/com/example/a17255/a8project/GamePlay.java

```diff
@@ -92,26 +160,173 @@ public MyCanvas(Context context) {
+                    score -= 7;
+
+                    currentSpeedX = Math.abs(currentSpeedX) - 1000;
+                    currentSpeedY = Math.abs(currentSpeedY) - 1000;
+                    obstaclePaint4.setColor(Color.CYAN);
+                }
+                if (Math.pow(mCurrentX - obstacleX5, 2) + Math.pow(mCurrentY - obstacleY5, 2) <=
+                        Math.pow(BALL_RADIUS + OBSTACLE_RADIUS_2, 2)) {
+//                    reduce score when collide
+                    score -= 7;
+
+                    currentSpeedX = Math.abs(currentSpeedX) - 1000;
+                    currentSpeedY = Math.abs(currentSpeedY) - 1000;
+                    obstaclePaint5.setColor(Color.CYAN);
+                }
+                if (Math.pow(mCurrentX - obstacleX3, 2) + Math.pow(mCurrentY - obstacleY3, 2) >
+                        Math.pow(BALL_RADIUS + OBSTACLE_RADIUS_2, 2)) {
+                    obstaclePaint3.setColor(Color.DKGRAY);
+                }
+                if (Math.pow(mCurrentX - obstacleX4, 2) + Math.pow(mCurrentY - obstacleY4, 2) >
+                        Math.pow(BALL_RADIUS + OBSTACLE_RADIUS_2, 2)) {
+                    obstaclePaint4.setColor(Color.DKGRAY);
+                }
+                if (Math.pow(mCurrentX - obstacleX5, 2) + Math.pow(mCurrentY - obstacleY5, 2) >
+                        Math.pow(BALL_RADIUS + OBSTACLE_RADIUS_2, 2)) {
+                    obstaclePaint5.setColor(Color.DKGRAY);
+                }
+
+
+//                detect the collision between the ball and the target
+                if (Math.pow(mCurrentX - targetX, 2) + Math.pow(mCurrentY - targetY, 2) <=
+                        Math.pow(BALL_RADIUS + TARGET_RADIUS, 2)) {
+
+//                    change score when collide
+                    currentSpeedX = 0;
+                    currentSpeedY = 0;
+                    score += 200;
+
+//                    save score in the SharedPreferences
+                    saveScoreInSharedPreferences(score);
+
+//                    alert dialog
+                    confirmFireMissiles();
+
+
+
+                }
+
+
+
+
+
+
+//                display score
+                drawScore(canvas);
+
+
+//                draw ball after fling
+
+                mCurrentX = mCurrentX + currentSpeedX / 4000 * DURATION_PER_FRAME_IN_MS;
+                mCurrentY = mCurrentY + currentSpeedY / 4000 * DURATION_PER_FRAME_IN_MS;
                 drawBall(canvas);
                 postInvalidateDelayed(DURATION_PER_FRAME_IN_MS);
+
             }
         }
 
+        private void drawScore(Canvas canvas) {
+            canvas.drawText(Integer.toString(score), SCORE_LOCATION_X, SCORE_LOCATION_Y, paint);
+        }
+
         private void drawBall(Canvas canvas) {
             canvas.drawCircle(mCurrentX, mCurrentY, BALL_RADIUS, paint);
         }
 
-        @Override
-        public boolean onTouchEvent(MotionEvent event) {
+        private void drawObstacle1(Canvas canvas, float x, float y, Paint paint) {
```

Comment:
👍 for creating a method for it, but better to rename it to something like drawLightGreyObstacle